agent/api/task/task_windows.go

```diff
@@ -253,3 +256,38 @@ func (task *Task) BuildCNIConfigAwsvpc(includeIPAMConfig bool, cniConfig *ecscni
 func (task *Task) BuildCNIConfigBridgeMode(cniConfig *ecscni.Config, containerName string) (*ecscni.Config, error) {
 	return nil, errors.New("unsupported platform")
 }
+
+// requiresCredentialSpecResource returns true if at least one container in the task
+// needs a valid credentialspec resource
+func (task *Task) requiresCredentialSpecResource() bool {
+	for _, container := range task.Containers {
+		if container.RequiresCredentialSpec() {
+			return true
+		}
+	}
+	return false
+}
+
+// initializeCredentialSpecResource builds the resource dependency map for the credentialspec resource
+func (task *Task) initializeCredentialSpecResource(config *config.Config, credentialsManager credentials.Manager,
+	resourceFields *taskresource.ResourceFields) error {
+	credspecContainerMapping := task.getAllCredentialSpecRequirements()
+	credentialspecResource, err := credentialspec.NewCredentialSpecResource(task.Arn, config.AWSRegion, task.ExecutionCredentialsID,
+		credentialsManager, resourceFields.SSMClientCreator, resourceFields.S3ClientCreator, credspecContainerMapping)
+	if err != nil {
+		return err
+	}
+
+	task.AddResource(credentialspec.ResourceName, credentialspecResource)
+
+	// for every container that needs credential spec vending, it needs to wait for all credential spec resources
+	for _, container := range task.Containers {
+		if container.RequiresCredentialSpec() {
+			container.BuildResourceDependency(credentialspecResource.GetName(),
+				resourcestatus.ResourceStatus(credentialspec.CredentialSpecCreated),
+				apicontainerstatus.ContainerCreated)
+		}
+	}
+
+	return nil
+}
```

Comment:
The changes in ecscni package are good, and should be done anyway.

But I am not in favor of duplicating code between our two supported OS platforms (Linux and Windows) to make the agent compile on an unsupported platform (Mac). We shouldn't be running our tests on Mac anyway since that is not a target platform, has different behaviors and hence the tests results may be misleading.

Reply:
thanks Onur, 

the `initializeCredentialSpecResource` should return an explicit error for unsupported platforms, like -
```
errors.New("unsupported platform")
```

this seems to be missed when support for gmsa on linux was added. with this PR, we achieve that, although the task_linux and task_windows implementations are the same.

any ideas on how to avoid that?

Reply:
One thing you could do is do a cross platform compile for osx in gh actions:

`GOOS=darwin go build ./` or something like that... you don't need osx hardware to do that but you still have some validation that your code isn't breaking on mac. You can also compile the tests themselves if you want bonus points: https://stackoverflow.com/questions/72721580/golang-compile-all-tests-across-a-repo-without-executing-them

I think we got rid of the mac tests a million years ago because freemium CI solutions tend to have limited OSX queues.

Reply:
thanks! I'll close this PR for now.